### PR TITLE
Refactor generate settings

### LINKED ISSUE #135: Update generateSettings to handle data mappings at any position in settings
Right now `generateSettings()` only handles data mappings at level 1 or 2 of the settings next - throws [this error](https://github.com/ASA-DIA-InteractiveSafetyGraphics/safetyGraphics/pull/128/files/244e68616f762c0dcb21af4f9328ac1c37243a43..3ce796650f8c10017070c5d32b7862361f2418b4#diff-a1dc3489ed58a9da5083928f6b0347efR92) otherwise. We should write a new (probably recursive) method to make this work at any level of the settings object. It will be a bit like the code for `getSettingValues()`.

### PR DESCRIPTION
**Pull request by jwildfire**
Update/refactor/simplify generate settings to deal with field level data mappings and prep for supporting multiple charts .

Closes #135 #151

### CODE REVIEW

File: R/generateSettings.R [hunk 1]
```diff
@@ -30,26 +31,38 @@
...
+    filter(.data$column_name != '')
+    
+    if(partial){
+      dataMappings<-dataMappings%>%filter(.data$text_key %in% partial_keys) 
+    }
+  }
   
-  settings<-list(
+  # build shell settings for each chart (move these to /data eventually?)
+  shells<-list()
```
Comment (pburnsdata) [inline R/generateSettings.R]: and maybe you only need to write the shell that matches the chart setting (so don't have to make a big list of them) - but that's not really a concern right now

File: R/setSettingsValue.R [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  stopifnot(
+    typeof(settings)=="list"
+  )
+  
+  firstKey <- key[[1]]
+  if(length(key)==1){
+    settings[[firstKey]]<-value
+    return(settings)
+  }else{
+    settings[[firstKey]]<-setSettingsValue(settings = settings[[firstKey]],key = key[2:length(key)], value)
```
Comment (pburnsdata) [inline R/setSettingsValue.R]: this is super cool - I'm a fan

File: R/generateSettings.R [hunk 3]
```diff
@@ -78,42 +91,12 @@ generateSettings <- function(standard="None", chart="eDish", partial=FALSE, part
...
-    }
-    
-  } else {
-    
-    settings <- potential_settings
-    
+  # loop through dataMappings and apply them to the shell
+  if(standard != "none"){
+    for(row in 1:nrow(dataMappings)){
+      shells[[chart]]<-setSettingsValue(settings = shells[[chart]], key = textKeysToList(dataMappings[row,"text_key"])[[1]], value = dataMappings[row, "column_name"])
```
Comment (pburnsdata) [inline R/generateSettings.R]: so much cleaner lolol

File: R/setSettingsValue.R (hunk 2 above)
Comment (pburnsdata) [inline R/setSettingsValue.R]: this is super cool - I'm a fan

File: R/generateSettings.R (hunk 1 above)
Comment (jwildfire) [reply R/generateSettings.R]: Filed #168 to cover this work.

**Pull request merged**